cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -154,3 +145,110 @@ func triggerPull(repositoryName string, pullConnectionDetails abaputils.Connecti
 	uriConnectionDetails.URL = body.Metadata.URI + expandLog
 	return uriConnectionDetails, nil
 }
+
+func checkPullRepositoryConfiguration(options abapEnvironmentPullGitRepoOptions) error {
+	if len(options.RepositoryNames) > 0 && len(options.RepositoryNamesFiles) > 0 {
+		log.Entry().Info("It seems like you have specified both the repositories to be pulled as an in-line configuration as well as in the dedicated repositories configuration file.")
+		log.Entry().Info("Please note that in this case the dedicated repositories configuration file will be handled with priority.")
+	}
+	if len(options.RepositoryNames) == 0 && len(options.RepositoryNamesFiles) == 0 {
+		return fmt.Errorf("Checking configuration failed: %w", errors.New("You have not specified any repository configuration to be pulled into the ABAP Environment System. Please make sure that you specified the repositories that should be pulled either in a dedicated file or via in-line configuration. For more information please read the User documentation"))
+	}
+	return nil
+}
+
+func pullReposFromFileConfig(repositoriesFilesConfig []string, pullConnectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, pollIntervall time.Duration) (err error) {
```

Comment:
With those two functions "pullReposFromFileConfig" and "pullReposFromConfig" there are a lot of duplications, which IMO leads to not well maintainable code. And essentially, those two functions are doing the same thing. 

A better approach would be to combine both sources of repositories into one list and then loop over this one list. (see https://github.com/SAP/jenkins-library/blob/master/cmd/abapEnvironmentCloneGitRepo.go)

Reply:
Re-factored these two functions for both steps. Can you review again?